### ISSUE TITLE
(Ledger client) Capture more details when linking anonymous wallet to custodian

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#15257]
### Overview of process
1. Person has Brave Rewards enabled
2. Uphold account is setup and verified
3. Person clicks the `Verify Wallet` button on brave://rewards
4. Login is made via Uphold (and when available, other custodian)
5. Backend calls are made; goal is to move funds from anonymous wallet to custodian (ex: Uphold)

### Problem
The problem is that sometimes folks are getting stuck and need to reach out to @brave/rewards-support to provide more information to help get unstuck. This is a bad experience and requires a lot of manual intervention

### What happens in the code
When a person w/ Brave Rewards enabled connects their wallet to Uphold for the first time, a few things will happen:
1. Check is made to make sure Uphold account is verified ([click here for code](https://github.com/brave/brave-core/blob/1cb5818aa0b70666c6aeea5ea9c06cc4e712171a/vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_wallet.cc#L64-L114))
2. If verified, a call is made to our backend to start the "claim". This will link the account ([click here for code](https://github.com/brave/brave-core/blob/1cb5818aa0b70666c6aeea5ea9c06cc4e712171a/vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_uphold/post_claim_uphold.cc#L138-L153))
3. If claim call is successful, the transfer tokens call is made ([click here for code](https://github.com/brave/brave-core/blob/1cb5818aa0b70666c6aeea5ea9c06cc4e712171a/vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc#L145-L147))

### Problems / what could be better
We throw away most of the response in steps 2 and 3 above (claim / transfer tokens). This could be extremely useful in debugging and could be logged to brave://rewards-internals or saved to profile

1. For Claim, we're only looking at the HTTP response code (`response.status_code`). We completely ignore the other fields, like `response.body` which have more information about the call ([click here for source](https://github.com/brave/brave-core/blob/1cb5818aa0b70666c6aeea5ea9c06cc4e712171a/vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_uphold/post_claim_uphold.cc#L158-L159))
2. For transfer tokens, we are given a `drain_id`. We should be able to poll this `drain_id` for a text status (ex: pending, in progress, delayed, complete). [iOS recently implemented a feature like this for QR code transfer to desktop](https://github.com/brave/brave-ios/issues/3162). You can see where the callback for transfer tokens resolves, it passes in `drain_id` which can then be logged to brave://rewards-internals or saved to profile ([click here for source](https://github.com/brave/brave-core/blob/1cb5818aa0b70666c6aeea5ea9c06cc4e712171a/vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc#L145-L147))

Besides the support cost and impact to customers... code-maintenance-wise, the client code can be a bit hard to follow (lots of callbacks!). This would be a great example to clean up

**Comment by bsclifton** [comment]
Some logging information has been captured with https://github.com/brave/brave-core/pull/8598 (solving https://github.com/brave/brave-browser/issues/15500)

**Comment by bsclifton** [comment]
We should definitely log `drain_id` if possible; also the text errors logged could be more specific / helpful as we're now logging them with https://github.com/brave/brave-core/pull/8598

**Comment by Miyayes** [comment]
I think the issues in this have been resolved with @szilardszaloki's overall work and updates to Rewards + Uphold states, so will close this out :)
